helix-core/src/main/java/org/apache/helix/task/TaskDriver.java

```diff
@@ -850,6 +858,56 @@ public static JobConfig getJobConfig(HelixManager manager, String job) {
     return workflowConfigMap;
   }
 
+  /**
+   * Batch get the configurations of all workflows in this cluster
+   * within the specified timeout in milliseconds.
+   *
+   * @param timeout a long integer presents the timeout, in milliseconds
+   * @return a map of <String, WorkflowConfig>
+   * @throws InterruptedException if the future thread was interrupted
+   * @throws ExecutionException if the future task completed exceptionally
+   * @throws TimeoutException if waiting for result timed out
+   */
+  public Map<String, WorkflowConfig> getWorkflows(long timeout)
```

Comment:
So the current getWorkflows() should be a special case of this call?
Can we replace the original method using this one? So as to avoid duplicate code.

Reply:
I had a conversation to discuss about the exception handling with Hunter. He suggested keeping the code as is. This is why I did not change the current code in getWorkflows().
If we would like to change that API, we could just do this getWorkflows() calls getWorkflows(0L). timeout = 0L means wait with no timeout. What do you think?

Reply:
I prefer getWorkflows(Integer.MAX_VALUE). So that the exception problem is somewhat resolved? Not sure about if you have other concerns.
But IMHO, very limited thing can be worse than duplicate code : )